src/lib/mongoBackend/mongoListSubscriptions.cpp

```diff
@@ -23,42 +23,171 @@
 * Author: Orion dev team
 */
 
+#include "common/sem.h"
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
 #include "mongoBackend/mongoListSubscriptions.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "mongoBackend/collectionOperations.h"
+
+#include "mongo/client/dbclient.h"
+
+using namespace ngsiv2;
+
+/* ****************************************************************************
+*
+* setSubscriptionId -
+*/
+extern void setSubscriptionId(Subscription* s, const BSONObj& r)
+{
+  s->id = r.getField("_id").OID().toString();
+}
+
+
+/* ****************************************************************************
+*
+* setSubject -
+*/
+extern void setSubject(Subscription* s, const BSONObj& r)
+{
+  // Entities
+  std::vector<BSONElement> ents = r.getField(CSUB_ENTITIES).Array();
+  for (unsigned int ix = 0; ix < ents.size(); ++ix)
+  {
+    BSONObj ent           = ents[ix].embeddedObject();
+    std::string id        = ent.getStringField(CSUB_ENTITY_ID);
+    std::string type      = ent.getStringField(CSUB_ENTITY_TYPE);
+    std::string isPattern = ent.getStringField(CSUB_ENTITY_ISPATTERN);
+
+    EntID en;
```

Comment:
Becasue in the current position it is more "tied" with the if/else that fills the variable.